### ISSUE TITLE
Notification enum value collision

### ISSUE THREAD

**Issue by Noshyaar** [godotengine/godot#12304]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Godot 3 8265ff4

**Issue description:**

Expected: notification enums are unique
Happened: 

|    | Type       | Name                                  |
|----|------------|---------------------------------------|
| 29 | Spatial    | NOTIFICATION_TRANSFORM_CHANGED        |
| 29 | CanvasItem | NOTIFICATION_TRANSFORM_CHANGED        |
| 41 | Control    | NOTIFICATION_MOUSE_ENTER              |
| 41 | Spatial    | NOTIFICATION_ENTER_WORLD              |
| 42 | Control    | NOTIFICATION_MOUSE_EXIT               |
| 42 | Spatial    | NOTIFICATION_EXIT_WORLD               |
| 43 | Control    | NOTIFICATION_FOCUS_ENTER              |
| 43 | Spatial    | NOTIFICATION_VISIBILITY_CHANGED       |
| 50 | Container  | NOTIFICATION_SORT_CHILDREN            |
| 50 | Skeleton   | NOTIFICATION_UPDATE_SKELETON          |

**Comment by akien-mga** [comment]
I mentioned something similar to @reduz a while ago, but he said that since those are not in the same inheritance branch, it doesn't matter. You won't get a Control and a Spatial notification at the same time, as you can only inherit from one of those.

**Comment by Zylann** [comment]
What if you have a control child of spatial in the tree? (Or it doesn't work with tree?)

**Comment by akien-mga** [comment]
The Control child will only receive notifications that match its type, regardless of its parent nodes.

**Comment by Noshyaar** [comment]
Agreed. There shouldn't be a problem unless you start doing `NOTIFICATION_SORT_CHILDREN ==  NOTIFICATION_UPDATE_SKELETON` for whatever reason :P

(Also that ghost was me)
